xmlrpc/man/update-tcospasswd.1
.TH update-tcospasswd 1 "Jun 21, 2006" "update-tcospasswd man page"

.SH NAME
update-tcospasswd \- TCOS util to generate crypt TCOS passwords files

.SH SYNOPSIS
.B update-tcospasswd
.br

.SH DESCRIPTION

This manpage show options for 
.B update-tcospasswd
script.

.PP
\fBupdate-tcospasswd\fP is a small shell script that create TCOS crypt 
password files with some questions.


.SH BUGS
Contact Mario Izquierdo <mariodebian@gmail.com>

Homepage: http://www.tcosproject.org

.SH SEE ALSO
tcos.conf(5)

.SH AUTHOR
Mario Izquierdo (mariodebian) <mariodebian@gmail.com>


debian/changelog
initramfs-tools-tcos (0.89.53) unstable; urgency=low

  * move some tcos-bottom script later
  * scripts/tcos-bottom/80shell: stop bootchart if exists
  * scripts/tcos-bottom/57devices: test ltspfsd
  * scripts/tcos-bottom/55mount_listener: only start if USB and IDE is
    enabled
  * set chmod +x in scripts
  * bin/tcos-genbootchart: use DISPLAY if not arg is passed
  * set 2.6.32-trunk-486 for Debian unstable
  * conf/base.conf: add TCOS_USBIP new var, disabled by default
  * hooks-addons/19usbip: use new var TCOS_USBIP
  * debian/tcos-standalone.init: provides tcos-standalone
  * debian/copyright: replace (C) with ©
  * debian/control: add misc:Depends and fix typo
  * xmlrpc/man/update-tcospasswd.1 tcos-standalone/tcos-standalone.1
    bin/tcos-gdm-autologin.1 bin/tcos-buildchroot.1 bin/gentcos.1: split
    lines longer than 80 chars
  * Makefile conf/tcos.conf.light: Add new light template run in 46 MB
  * conf/tcos-run-functions.sh: update list of killed apps
  * tcos-standalone/tcos-standalone.desktop: remove obsolete Encoding
  * New XMLRPC method: config, read vars from tcos.conf in thin client

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Fri, 18 Dec 2009 19:04:14 +0100

initramfs-tools-tcos (0.89.52) unstable; urgency=low

  * Enable/disable composite Xorg extension
  * fix if::else TCOS_COMPOSITE
  * downgrade some depends to recommends
  * cosmetic hanges in standalone init script
  * bin/tcos-buildchroot: copy /etc/localtime /etc/adjtime if exists
  * hooks-addons/50xorg: copy linked libs
  * add TCOS_CPUFREQ support with powersave scaling governor
  * tcos-standalone/tcos-standalone.py replace obsolete popen2 with
    subprocess
  * scripts/tcos-bottom/30devices: disable -a in ltspfsd, not needed
    anymore
  * remove rtptools and sox depends, audio multicast use pulseaudio
    native RTP module
  * fix typo suggest => suggests

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sun, 13 Dec 2009 19:24:56 +0100

initramfs-tools-tcos (0.89.51) unstable; urgency=low

  * new bin waitforX, to e used in startup scripts
  * copy waitforX and use it instead of 'pidof Xorg'
  * dnsmasq by default
  * detect VGA vendor with new script get_vga_vendor
  * update USBIP bins PATH
  * add 2 new openchrome cards
  * hooks-addons/53mga: fix typo
  * hooks-addons/21pulseaudio: fix typo
  * tcos-genbootchart: move to /usr/bin
  * hooks-addons/05devices call ltspfs-set-xprop in X11/PreRun
  * Update copyright
  * fix path in tcos-genbootchart

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Mon, 07 Dec 2009 21:37:52 +0100

initramfs-tools-tcos (0.89.50) unstable; urgency=low

  * Depend on ltspfsd-core and ltspfs
  * Use evdev as keyboard driver when kbd is not found
  * put XbdModel=abnt2 in evdev driver
  * update depends, change to i386 the thin client depends
  * update kernels
  * revert arch depends

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sun, 22 Nov 2009 20:14:37 +0100

initramfs-tools-tcos (0.89.49) unstable; urgency=low

  * remove libnss-db, new libc conflicts with it
  * set home if getent fails
  * bin/tcos-buildchroot: remove libnss-db from PACKAGES
  * hooks-addons/00main: copy libnss_db.so.2 if exists
  * bin/get_xauth: read cookie from GDM/KDM when no user is logged
  * scripts/tcos-bottom/10ldconfig: export env vars in /root/.profile
  * scripts/tcos-bottom/80shell: change order in PS1 var
  * debian/initramfs-tools-tcos.postinst: fix typo

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sun, 08 Nov 2009 14:57:29 +0100

initramfs-tools-tcos (0.89.48) unstable; urgency=low

  * fix startscale STANDALONE auth argument
  * copy localtime and adjtime from system to chroot
  * save date in BIOS after run ntpdate and copy /etc/services always,
    not only if inetd is ENABLED
  * bin/get_filesystem: call udevadm if exists
  * bin/tryXorg.sh: fix AutoAddDevices in xorg.conf.auto

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Tue, 13 Oct 2009 12:02:21 +0200

initramfs-tools-tcos (0.89.47) unstable; urgency=low

  * fix typo and use DPATH inseat of reserved var PATH
  * udev/mount_listener.c don't show errors if poll() return 0
  * hooks-addons/03udev: copy /lib/udev/rules.d/40-alsa.rules, 
    Ubuntu Jaunty fix sound
  * add support to update BIOS date/time with ntpdate, disabled by default
  * use blkid, from util-linux (if avalaible) to get filesystem type
  * call user scripts in /etc/tcos/pxelinux.d/*.hook files
  * log ntpdate in initramfs.debug file
  * fix typos in example install
  * add new files to initramfs package
  * check for reopen in x11vnc before using it
  * fix exec pxelinux.d hooks and fix example hack

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sun, 04 Oct 2009 15:43:59 +0200

initramfs-tools-tcos (0.89.46) unstable; urgency=low

  * pulseaudio: set exit-idle-time to -1, only set SHM if low memory <
    64MiB, and don't copy udev or x11 modules
  * create p910nd lock dir
  * update Ubuntu Jaunty kernel to 2.6.28-15-generic
  * don't run vol_id if not found, and add udevinfo detection method
  * udev/tcos-udevd.sh udev/save-udev.sh: translate DEVPATH to old short
    one

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Thu, 01 Oct 2009 16:03:36 +0200

initramfs-tools-tcos (0.89.45) unstable; urgency=low

  * Allow to start x11vnc server in scale mode
  * Include dbus utils, disabled by default
  * PulseAudio revert to deprecated module-detect add users and revert
    module-x11-publish => pax11publish
  * Remove chroot package from DISTRO_PACKAGES

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Wed, 30 Sep 2009 22:41:29 +0200

initramfs-tools-tcos (0.89.44) unstable; urgency=low

  * Use abnt2 keymap for Brasil users
  * Add support for NTFS-3G
  * Add more info about dnsmasq and gPXE
  * Allow to personalize usplash resolution
  * Fix rdesktop IP detection and use correct keymap
  * xmlrpc/*:
    - Add support to get screensize in info.[ch]


xmlrpc/sh/update-tcospasswd.sh
#!/bin/bash

#/*
#* update-tcospasswd.sh shell script to generate tcospasswd
#*                      to PXES, LTSP, TCOS or STANDALONE
#* Copyright (C) 2006,2007,2008  mariodebian at gmail
#*
#* This program is free software; you can redistribute it and/or
#* modify it under the terms of the GNU General Public License
#* as published by the Free Software Foundation; either version 2
#* of the License, or (at your option) any later version.
#*
#* This program is distributed in the hope that it will be useful,
#* but WITHOUT ANY WARRANTY; without even the implied warranty of
#* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#* GNU General Public License for more details.
#*
#* You should have received a copy of the GNU General Public License
#* along with this program; if not, write to the Free Software
#* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#*/



if [ "$(whoami)" != "root" ]; then
  echo "Error: Need to exec as root user."
  exit 1
fi

_is_ltsp=$(echo /opt/ltsp*)
_is_pxes=$(echo /opt/pxes*)
_is_tcos=$(echo /etc/tcos)
_username=""
_passwd=""
_arq=""
_fpass=""
_silent=0
_noencrypted=0
_secret=0

if [ -d ${_is_ltsp} ]; then
  ltsp=1
fi
if [ -d ${_is_pxes} ]; then
  pxes=1
fi
if [ -d ${_is_tcos} ]; then
  tcos=1
fi

for x in $@; do
    case $x in
	--user=*)
		_username="${x#--user=}"
		;;
	--passwd=*)
		_passwd="${x#--passwd=}"
		;;
	--arq=*)
		_arq="${x#--arq=}"
		;;
	--file=*)
		_fpass=${x#--file=}
		;;
	--silent)
		_silent=1
		;;
	--noencrypted)
		_noencrypted=1
		;;
	--tcosmonitor-secret)
		_secret=1
		;;
    esac
done


make_passwd(){
$(which tcospasswd) $1
}


if [ -z $_arq ] && [ -z $_fpass ] && [ $_secret -eq 0 ]; then
   echo "Arquitecture to update password:"
   echo ""
   if [ ${ltsp} ]; then
     echo "  * LTSP"
   fi

   if [ ${pxes} ]; then
     echo "  * PXES"
   fi

   if [ ${tcos} ]; then
     echo "  * TCOS"
   fi


   echo "  * STANDALONE"

   echo -n "              : "
   read _arq

   if [ "$_arq" = "LTSP" ]; then
     _fpass=/opt/ltsp/i386/etc/tcospasswd

   elif [ "$_arq" = "PXES" ]; then
     _fpass=$(echo /opt/pxes*)/stock/dist/etc/tcospasswd

   elif [ "$_arq" = "TCOS" ]; then
     _fpass=/etc/tcospasswd

   elif [ "$_arq" = "STANDALONE" ]; then
     _fpass=/etc/tcospasswd
   else
     echo "Error: Arquitecture: ${_arq} not supported, please write LTSP, PXES, TCOS or STANDALONE"
     exit 1
   fi
fi


if [ -z $_username ];then
  echo -n "Username (usually root): "
  read _username
fi

if [ -z $_passwd ];then
  echo -n "TCOS Password: "
  read _passwd
fi

if [ $_secret -eq 1 ];then
  _fpass=/etc/tcos/secrets/tcosmonitor-secret
  [ ! -d $(dirname ${_fpass}) ] && exit 0
  echo "$_username:$_passwd" > ${_fpass}
  chown -f root:tcos ${_fpass}
  chmod -f 640 ${_fpass}
  exit 0
fi

if [ $_noencrypted -eq 0 ];then
  _passwd="$(make_passwd $_passwd)"
fi

string="$_username:$_passwd"

if [ $_silent -eq 0 ];then
  if [ ! -d $(dirname ${_fpass}) ]; then 
    echo "Error: $(dirname ${_fpass}) don't exists."
    exit 1
  fi

  echo -n "I will write this file:
    ${_fpass}
            Continue? [Y/n] "
  read _ans

  if [ "${_ans}" != "n" ]; then
    echo ${string} > ${_fpass}
    chown -f root:tcos ${_fpass}
    chmod -f 640 ${_fpass}
    echo ""
    echo "In case of PXES or TCOS (not STANDALONE), please update the"
    echo "boot image with pxesconfig or gentcos."
    echo "Done."
    exit 0
  fi
else
  echo ${string} > ${_fpass}
  chown -f root:tcos ${_fpass}
  chmod -f 640 ${_fpass}
  exit 0
fi

echo "Aborted."
exit 1

